components/Ripple/src/private/MDCRippleLayer.m

```diff
@@ -58,6 +58,9 @@ - (void)startRippleAtPoint:(CGPoint)point
   self.opacity = 1;
   self.position = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
   if (!animated) {
+    if (completion) {
+      completion();
+    }
```

Comment:
I'd expect this to be called after the delegate; did you place it before intentionally?

Reply:
It's based on the order a bit more down the file if `animated = YES`:

```
[CATransaction setCompletionBlock:^{
  self->_startAnimationActive = NO;
  if (completion) {
    completion();
  }
  [self.rippleLayerDelegate rippleLayerTouchDownAnimationDidEnd:self];
}]
```

In there we are also calling the completion block before. I'm happy to change both as I would assume the same.